### PR TITLE
reflect CLI options on web URL

### PR DESCRIPTION
**Pull request by thisiscam**
Attempt at #50 .

Currently serializes CLI options with JSON, which is totally unreadable.

### CODE REVIEW

File: docs/index.ts [hunk 1]
```diff
@@ -198,6 +199,60 @@ function getOptions(): Options {
...
+function updateURLParams() {
+	const options = getOptions();
+	const options_json = JSON.stringify(options);
+	window.history.pushState(options, "", `index.html?cli=${encodeURIComponent(options_json)}`);
+}
+
+function getOptionsFromURL() : Options {
+	const queryString = window.location.search;
+	const urlParams = new URLSearchParams(queryString);
+	const options_json = urlParams.get('cli');
```
Comment (FlamingTempura) [inline docs/index.ts:249]:
```suggestion
	const options_json = urlParams.get('opt');
```

File: docs/index.ts [hunk 2]
```diff
@@ -198,6 +199,60 @@ function getOptions(): Options {
...
+	options.lowercase.checked = opts.lowercase;
+	options.generateKeys.checked = opts.generateKeys;
+	options.maxAuthors.checked = opts.maxAuthors !== null;
+	options.maxAuthorsNum.value = String(opts.maxAuthorsNum || 0);
+}
+
+function updateURLParams() {
+	const options = getOptions();
+	const options_json = JSON.stringify(options);
+	window.history.pushState(options, "", `index.html?cli=${encodeURIComponent(options_json)}`);
```
Comment (FlamingTempura) [inline docs/index.ts:243]:
```suggestion
	window.history.pushState(options, "", `index.html?opt=${encodeURIComponent(options_json)}`);
```

**Review by FlamingTempura — CHANGES_REQUESTED** [review]
Thanks for the PR!

The only thing I'd like changed is to call the query param `opt` instead of `cli` (since it's not really options specific to the command line).

Other than that, good to go! Really appreciate you working on this feature (I don't find much time to maintain this library).

I'll make the merge when this change is made.

File: docs/index.ts (hunk 2 above)
Comment (thisiscam) [reply docs/index.ts]: Fixed in the new commit. Thanks!

**Comment by thisiscam** [comment]
@FlamingTempura I implemented your suggestion and added a delay to textarea inputs, to avoid accumulating a browser history time frame for each character one type.

**Comment by FlamingTempura** [comment]
Merged, thanks for your contribution

**Pull request merged**